libp2p/protocols/secure/secure.nim

```diff
@@ -11,15 +11,27 @@ import options
 import chronos, chronicles
 import ../protocol,
        ../../stream/streamseq,
-       ../../connection,
+       ../../stream/connection,
        ../../peerinfo
 
 type
   Secure* = ref object of LPProtocol # base type for secure managers
 
   SecureConn* = ref object of Connection
+    stream*: Connection
     buf: StreamSeq
 
+method initStream*(s: SecureConn) =
+  if s.objName.len == 0:
+    s.objName = "SecureConn"
+
+  procCall Connection(s).initStream()
+
+method close*(s: SecureConn) {.async.} =
+  await s.stream.close()
```

Comment:
is stream surely not nil?

Reply:
good point, you never know with async, so good idea to guard against it.